### PR TITLE
Test ApplyConfState after restart

### PR DESCRIPTION
**Pull request by serathius**
This test was written to show issue with https://github.com/etcd-io/etcd/pull/17019.
We cannot disable validation on entries before v3 state because we are still using them to update confState, which is still loaded from snapshot and not v3.

### CODE REVIEW

File: server/etcdserver/server_test.go [hunk 1]
```diff
@@ -198,6 +198,115 @@ func TestV2SetClusterVersion(t *testing.T) {
...
+				NodeID: nodeID,
+			}),
+		},
+		{
+			Term:  1,
+			Index: 3,
+			Type:  raftpb.EntryConfChange,
+			Data: pbutil.MustMarshal(&raftpb.ConfChange{
+				Type:    raftpb.ConfChangeAddNode,
+				NodeID:  nodeID,
```
Comment (ahrtr) [inline server/etcdserver/server_test.go:239]: Note that etcd generates a new memberId each time when users add a new member. It isn't valid to use the same nodeID.

Comment (serathius) [reply server/etcdserver/server_test.go:239]: True, updated to use ConfChangeUpdateNode

**Comment by serathius** [comment]
cc @ahrtr @fuweid @chaochn47 @siyuanfoundation

**Review by fuweid — APPROVED** [review]
LGTM

**Review by ahrtr — APPROVED** [review]

**Pull request merged**